tests/api/api/test_client_spec.py

```diff
@@ -124,3 +126,92 @@ def test_client_spec(
     )
     assert response_body["ce_mode"] == response_body["ce"]["mode"] == ce_mode
     assert response_body["ce"]["release"] == ce_release
+
+
+def test_client_spec_response_based_on_client_version(
```

Comment:
generally, I like it better when we use pytest parametrize because it's easier to follow which case failed and it makes the code shorter but of course this way is fine as well